src/client/dfs/dfs.c

```diff
@@ -3790,7 +3790,7 @@ readdir_int(dfs_t *dfs, dfs_obj_t *obj, daos_anchor_t *anchor, uint32_t *nr,
 			D_GOTO(out, rc = daos_der2errno(rc));
 
 		for (ptr = enum_buf, i = 0; i < number; i++) {
-			strncpy(dirs[key_nr].d_name, ptr, kds[i].kd_key_len);
+			memcpy(dirs[key_nr].d_name, ptr, kds[i].kd_key_len);
```

Comment:
Maybe check that `kds[i].kd_key_len < sizeof(dirs[key_nr].d_name)`? (I realize that this wasn't checked before, maybe it's guaranteed elsewhere)

Reply:
If that condition would be false, then we have landed some data corruption in the container and this would be the least of the problems :-)

Reply:
But if we don't detect it here, we'll have two data corruptions. :)

Reply:
my point was we probably won't even get here if there is data corruption in the container.
but sure, i can add that check if needed to repush. otherwise can add it in another PR im working on.

Reply:
Adding in another PR sounds good to me.

Reply:
Alternatively, if the check it's 100% redundant, a comment that explains this would help (it's not obvious unless you know the server side, I guess).

Reply:
I think there is an issue here but adding in another PR seems the way to go.